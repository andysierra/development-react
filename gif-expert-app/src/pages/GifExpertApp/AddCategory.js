import {useState} from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({setCategories}) => { 

  const [inputValue, setInputValue] = useState('');

  const onChange = ({target}) => setInputValue(target.value);

  const handleSubmit = e => {
    e.preventDefault();
    if(inputValue.trim().length > 2) {
      setCategories(cats => cats.includes(inputValue)? cats : [...cats, inputValue]);
      setInputValue('');
    }
  }

  return (
    <form onSubmit={handleSubmit}>
      <input
        type        = "text"
        className   = "form-control"
        value       = {inputValue}
        pattern     = "^.{3,}$"
        required    = {true}
        title       = "Debe tener mÃ­nimo 3 letras"
        placeholder = 'Buscar un GIF'
        onChange    = {onChange}/>
    </form>
  );

};

AddCategory.propTypes = {
  setCategories : PropTypes.func.isRequired
}

export default AddCategory;