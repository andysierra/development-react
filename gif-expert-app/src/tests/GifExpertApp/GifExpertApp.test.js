import { shallow } from "enzyme";
import GifExpertApp from "../../pages/GifExpertApp/GifExpertApp";

describe('Pruebas sobre GifExpertApp', ()=>{

  test('Debe de hacer match con snapshot', ()=>{
    const wrapper = shallow(<GifExpertApp/>);
    expect(wrapper).toMatchSnapshot();
  });

});