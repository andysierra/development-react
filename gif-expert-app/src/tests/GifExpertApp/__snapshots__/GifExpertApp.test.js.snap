// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre GifExpertApp Debe de hacer match con snapshot 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <div
    className="d-flex flex-column-reverse justify-content-start"
  />
</div>
`;
